worker-workflow/src/main/resources/add-failures.js

```diff
@@ -40,8 +40,14 @@ function addFailures (document, failures, extractSourceCallback, action) {
             ? extractSourceCallback(f)
             : getCurrentWorkerName(document) + " " + getCurrentWorkerVersion(document);
         var isWarningFlag = (typeof isWarning === 'function') ? isWarning(f): false;
+        var failureId = f.getFailureId();
+        if(failureId !== null) {
+            var indexOfWarning = failureId.indexOf("-WARNING");
```

Comment:
This is checking for `-WARNING` anywhere in the failure id.  It is only meant to be looking for it as a suffix.  Also for encapsulation purposes, finding this suffix should definitively mean that the failure is a warning, even without calling the `isWarning()` function.